backend/grpc-server/src/http/error.rs

```diff
@@ -0,0 +1,55 @@
+use axum::{
+    http::StatusCode,
+    response::{IntoResponse, Response},
+    Json,
+};
+use serde::Serialize;
+
+#[derive(Debug)]
+pub struct HttpError {
+    pub status: StatusCode,
+    pub message: String,
+}
+
+#[derive(Serialize)]
+struct ErrorResponse {
+    error: ErrorDetail,
+}
+
+#[derive(Serialize)]
+struct ErrorDetail {
+    message: String,
+    code: String,
+}
+
+impl IntoResponse for HttpError {
+    fn into_response(self) -> Response {
+        let body = Json(ErrorResponse {
+            error: ErrorDetail {
+                message: self.message.clone(),
+                code: format!("{}", self.status.as_u16()),
+            },
+        });
+        (self.status, body).into_response()
+    }
+}
+
+// Convert tonic::Status to HTTP error
+impl From<tonic::Status> for HttpError {
+    fn from(status: tonic::Status) -> Self {
+        let http_status = match status.code() {
+            tonic::Code::InvalidArgument => StatusCode::BAD_REQUEST,
+            tonic::Code::NotFound => StatusCode::NOT_FOUND,
+            tonic::Code::AlreadyExists => StatusCode::CONFLICT,
+            tonic::Code::PermissionDenied => StatusCode::FORBIDDEN,
+            tonic::Code::Unauthenticated => StatusCode::UNAUTHORIZED,
+            tonic::Code::Unimplemented => StatusCode::NOT_IMPLEMENTED,
```

Comment:
The `Unimplemented` code mapping to `NOT_IMPLEMENTED` (501) might not be the best choice. A 501 status means the server doesn't support the functionality. For gRPC Unimplemented, consider using 404 or 400 instead, as it typically means the method isn't available for this specific request rather than not being implemented at all.

```suggestion
            tonic::Code::Unimplemented => StatusCode::NOT_FOUND,
```